eth/stagedsync/stage_headers.go

```diff
@@ -423,7 +424,27 @@ func handleNewPayload(
 		log.Info(fmt.Sprintf("[%s] New payload missing parent", s.LogPrefix()))
 		cfg.hd.SetPoSDownloaderTip(headerHash)
 		schedulePoSDownload(requestId, header.ParentHash, headerNumber-1, s, cfg)
-		return &engineapi.PayloadStatus{Status: remote.EngineStatus_SYNCING}, nil
+		currentHeadNumber := rawdb.ReadCurrentBlockNumber(tx)
+		if currentHeadNumber != nil && math.GetAbsoluteDifference(*currentHeadNumber, headerNumber) < 32 {
+			// We try waiting until we finish downloading the PoS blocks if the distance from the head is enough,
+			// so that we will perform full validation.
+			for i := 0; cfg.hd.PosStatus() == headerdownload.Syncing && i < 10; i++ {
```

Comment:
This waiting loop should wrap the `cfg.hd.PosStatus() == headerdownload.Synced` check. Otherwise, there's no point in sleeping 10ms 10 times, we can just sleep 100ms straight away.

Reply:
oh yes right, you also have IDLE...